### PR TITLE
feat: implement abbreviated manifest

### PR DESCRIPTION
**Pull request by juanpicado**
- [x] implement https://github.com/verdaccio/verdaccio/issues/2792
  - [x] filter out versions with the abbreviated manifest 
- [x] add the extra field `time` field. Discussion: https://github.com/verdaccio/verdaccio/discussions/3315
- [x] tests
- [x] docs
- [x] minor improvements and bugs

### How to test

> Use pnpm 6 

```
pnpm install
pnpm build
pnpm debug
```

### CODE REVIEW

**Comment by juanpicado** [comment]
@zkochan this should be enough for start experimenting, let me know if you need more.

**Comment by zkochan** [comment]
Thank you! You're awesome!

File: packages/store/src/storage.ts [hunk 1]
```diff
@@ -523,17 +525,57 @@ class Storage {
...
+      },
+      {}
+    );
+    const convertedManifest = {
+      name: manifest['name'],
+      [DIST_TAGS]: manifest[DIST_TAGS],
+      versions: abbreviatedVersions,
+      modified: manifest.time.modified,
+      // NOTE: special case for pnpm https://github.com/pnpm/rfcs/pull/2
+      time: manifest.time,
```
Comment (zkochan) [inline packages/store/src/storage.ts:555]: For some reason I am getting an empty object. So it seems to not work.

Comment (zkochan) [reply packages/store/src/storage.ts:555]: Interesting, even in the full metadata the time is an empty object. Although if I request it from the registry, time is present.

Comment (juanpicado) [reply packages/store/src/storage.ts:555]: 🤔 ok might be something wrong, I've been refactoring the time field for this major version. How can I reproduce it so I can dive a bit more?.

Comment (zkochan) [reply packages/store/src/storage.ts:555]: Just request a metadata file, like `http://localhost:4873/is-positive` and see that the time is set to `{}`.

Comment (juanpicado) [reply packages/store/src/storage.ts:555]: Indeed you found a bug :) I'll dig and fix it.

Comment (juanpicado) [reply packages/store/src/storage.ts:555]: Ok, was a little typo https://github.com/verdaccio/verdaccio/pull/3322/commits/1c188ff0dc82621c321270420adb56381a092253#diff-37b316ba08d3e2164985c8e501dabb23e83ed42b4d999c4b5011730d019d58bfR1702 I'd increase test later, but now should works just fine.

Comment (zkochan) [reply packages/store/src/storage.ts:555]: thanks, now it works as expected.

**Pull request merged**